src/v3/src/transformer/terminal/transformTerminalMessages.ts

```diff
@@ -157,6 +157,23 @@ export const transformTerminalMessages: TerminalKeyTransformer = (transaction, f
     return formBag;
   } else if (containsMessageKey(TERMINAL_KEY.IDX_RETURN_LINK_OTP_ONLY, displayedMessages)) {
     return transformEmailMagicLinkOTPOnly(transaction, formBag);
+  } else if (containsMessageKey(TERMINAL_KEY.DEVICE_ACTIVATED, displayedMessages)) {
+    // Displays device activated terminal state title as a success InfoBox instead of a title
+    const deviceActivatedAsSuccessMessage: WidgetMessage = {
+      class: 'SUCCESS',
+      message: loc('device.code.activated.success.title', 'login'),
+      i18n: { key: 'device.code.activated.success.title' },
+    };
+    displayedMessages.unshift(deviceActivatedAsSuccessMessage);
+  } else if (containsOneOfMessageKeys([TERMINAL_KEY.DEVICE_NOT_ACTIVATED_CONSENT_DENIED,
+    TERMINAL_KEY.DEVICE_NOT_ACTIVATED_INTERNAL_ERROR], displayedMessages)) {
+    // Displays device not activated terminal states titles as an error InfoBox instead of a title
+    const deviceNotActivatedAsErrorMessage: WidgetMessage = {
+      class: 'ERROR',
+      message: loc('device.code.activated.error.title', 'login'),
+      i18n: { key: 'device.code.activated.error.title' },
+    };
+    displayedMessages.unshift(deviceNotActivatedAsErrorMessage);
```

Comment:
Not in scope but: Oh man, we really need to refactor the terminal message transformations 😅. This is getting to be a real mess.